### PR TITLE
[fix]: preserve bare-string tool call actions in streaming deep copy

### PR DESCRIPTION
**Pull request by citrocat**
## Summary

Follow-up to #7060, which taught `ResponsesToolMessageActionStruct` to decode bare-string actions (e.g. `image_generation_call`'s `"generate"`) into `ResponsesToolCallActionStr`. The Framework streaming accumulator's `deepCopyResponsesMessage` still copies only the object action variants, so a message carrying a bare-string action comes out of the copy with an all-nil action struct, and `MarshalJSON` on that copy fails with `responses tool message action struct is empty`.

This PR originally carried an independent fix for the decode side (#5834). That part is now covered by #7060, so the branch has been rewritten to contain only the remaining accumulator gap — a single small commit.

## Changes

- `framework/streaming/responses.go`: copy `ResponsesToolCallActionStr` in `deepCopyResponsesMessage`. Done by field name via the existing `copyOptionalStringFieldByName` helper (generalised to `any`) so the `framework` module keeps building against its pinned Core release, which predates the field.
- `framework/streaming/responses_test.go`: regression test `TestDeepCopyResponsesMessagePreservesBareStringAction`, asserting the value is preserved and not pointer-aliased. It skips when the pinned Core lacks the field.

## Type of change

- [x] Bug fix

## Affected areas

- [x] Framework (Go)

## How to test

```sh
# Against the local core (field present): must pass.
go work init ./core ./framework
cd framework && go test -race ./streaming/

# Against the pinned released core: framework must build, the new test skips.
cd framework && GOWORK=off go build ./... && \
  GOWORK=off go test ./streaming/ -run TestDeepCopyResponsesMessagePreservesBareStringAction -v
```

Without the `responses.go` change the new test fails with `bare-string action was not preserved`.

## Breaking changes

- [ ] Yes
- [x] No

Related: #5834, #7060

### CODE REVIEW

File: core/schemas/responses.go [hunk 1]
```diff
@@ -1803,10 +1804,29 @@ func (action ResponsesToolMessageActionStruct) MarshalJSON() ([]byte, error) {
...
+	// Image-generation actions are strings; the other action variants are objects.
+	if trimmed[0] == '"' {
+		var imageGenerationAction string
+		if err := Unmarshal(trimmed, &imageGenerationAction); err != nil {
+			return fmt.Errorf("failed to unmarshal image generation action: %w", err)
+		}
+		action.ResponsesImageGenerationCallAction = &imageGenerationAction
+		return nil
+	}
+
```
[the remark this answers is not in the document]
Comment (citrocat) [reply core/schemas/responses.go:2077]: Good catch — fixed in 195f35a56. `UnmarshalJSON` now rejects any non-string, non-object token (`null`, numbers, booleans, arrays) with a clear error before the object-dispatch peek, instead of letting `null` silently fall through to a zero-valued computer action. Added coverage for `null`/`42`/`true`/`[]` in `TestResponsesToolMessageActionStruct_EdgeCases`. Re-ran `go test -race ./core/schemas ./framework/streaming`, `./core/providers/openai`, plus the standalone `GOWORK=off` module builds — all green.

**Comment by citrocat** [comment]
@akshaydeo — flagging for your review. This is an upstream schema incompatibility, not a config or timeout issue, and it is currently live in our production deployment.

**The bug**: OpenAI Responses `image_generation_call.action` is a JSON string (`"generate"`), but `ResponsesToolMessageActionStruct.UnmarshalJSON` modeled every action as an object union with a `type` discriminator. When a valid image-generation event arrives, decoding fails and the streaming handler logs `Failed to parse stream response` and skips the **entire SSE frame** — including the terminal `response.completed`. Clients then hang until idle/client timeout or hit an abnormal EOF; they never receive a terminal event.

**Severity**: we observe this in production telemetry on `transports/v1.6.6` — ~16 occurrences on Aug 4, 2–4/day since, all `Mismatch type ... value "generate"`. Every occurrence is a lost terminal event for that request. The same object-only schema exists on the latest stable release, so anyone using image generation through the Responses API is affected. This is silent data loss: the request returns HTTP 200 with partial events and no terminal state.

**The fix** (195f35a56): accept string-valued actions, keep the object variants unchanged, reject non-string/non-object tokens, preserve the action through the Framework accumulator, and retain the optional request-side `action`. Minimal 5-file diff (+100/−6) with regression tests; `go test -race` on affected packages plus standalone module builds are green. CodeRabbit has approved.

Would appreciate your review when you have a chance.

**Comment by citrocat** [comment]
@TejasGhatte @akshaydeo — gentle follow-up on this one, no rush.

A quick data point that may help with prioritization: we have been running this exact patch in our production deployment for 6 days now (cherry-picked onto the v1.6.6 baseline we run). Results so far:

- **Before the patch**: the `Failed to parse stream response ... Mismatch type ... value "generate"` decode error occurred daily (2–16 occurrences/day over the preceding two weeks), each one dropping the terminal `response.completed` frame for an image-generation stream.
- **After the patch**: zero occurrences over the full 6 days, with normal image-generation traffic continuing throughout. No regressions observed on any other Responses traffic.

The PR itself is in a quiet state: single commit, rebased cleanly on current `dev` after last week's branch rework, CodeRabbit approved (its earlier null-token concern is addressed with a guard + edge-case tests), and all visible checks green. The remaining steps — human review and the approval-gated workflows — are on the maintainer side.

Happy to adjust anything if the review turns up concerns. Thanks!

**Comment by citrocat** [comment]
Rebased onto the latest `dev` (`1c52a9a0a`) after the recent history rewrite left this PR in a conflicting state — the branch is now a single clean commit again and GitHub reports it as mergeable.

Nothing changed in the patch itself: same 5 files, +100/−6, applied without conflicts. `core/schemas` and `core/providers/openai` tests pass; `framework/streaming` tests also pass when built against the local core via `go.work` (the framework module currently doesn't compile against its pinned released core on `dev` due to the unrelated `MCPClientConfig.AllowByDefault` addition, so that's pre-existing and not from this patch).

@akshaydeo @TejasGhatte — whenever you get a chance, this should now be straightforward to review. The patch is still running in our production deployment with no regressions.

**Comment by citrocat** [comment]
Superseded by #7067. The decode-side fix this PR originally carried landed upstream in #7060; the remaining Framework streaming deep-copy gap is now tracked in #7067 as a fresh single-commit PR.
